provider.go

```diff
@@ -0,0 +1,81 @@
+// Package namedotcom implements a DNS record management client compatible
+// with the libdns interfaces for namedotcom.
+
+package namedotcom
+
+import (
+	"context"
+	"github.com/libdns/libdns"
+)
+
+// Provider implements the libdns interface for namedotcom
+type Provider struct {
+	Client
```

Comment:
This probably doesn't need to be embedded/exported does it? Nothing within the Client type is exported, as far as I can tell.

Reply:
Thanks for reviewing!

It doesn't need to be exported. I have corrected this in the latest commit. (https://github.com/justenb/libdns/commit/2c725af50ba1f98b2d4ee322e1fc0eeb03ac306d)

I will follow up with pushing the latest to the libdns/namedotcom repo along with adding the caddy module.

Reply:
Awesome, thanks again for contributing this!